### ISSUE TITLE
Unnecessary `reportUnknownArgumentType` for nested list/dict literal

### ISSUE THREAD

**Issue by RobertCraigie** [microsoft/pyright#6721]
**Describe the bug**

When a function argument accepts any type (i.e. defined as `object`) then certain expressions will result in `reportUnknownArgumentType` being reported even though it doesn't matter that certain parts of the type are unknown because the type it's being assigned to is `object` and it's not used in a type variable or anything that would leak this type into anything else. Specifically, this happens when an empty dictionary literal is inside of a list.

**Code or Screenshots**

```py
def takes_anything(arg: object) -> None:
    ...

takes_anything([{}])  # error
takes_anything([])  # fine
takes_anything({})  # fine
```

Pyright reports this error in strict mode
```
error: Argument type is partially unknown
    Argument corresponds to parameter "arg" in function "takes_anything"
    Argument type is "list[dict[Unknown, Unknown]]" (reportUnknownArgumentType)
```

**VS Code extension or command-line**

CLI: `1.1.340`

If it's helpful, I was curious and bisected this error & it started happening in `1.1.282`

**Comment by erictraut** [comment]
Pyright is working as designed here.

All of the above cases arguably should generate a `reportUnknownArgumentType`, but pyright has special-case handling specifically for empty `list` and `dict` expressions, since those are commonly used. The special case doesn't extend to non-empty lists as in your first example.

Your options here are:
1. If you don't want to see "unknown argument type" errors in general, stick with the "standard" type checking rules rather than using "strict".
2. Use a temporary variable with a type annotation.
```python
x: list[dict[Any, Any]] = [{}]
takes_anything(x)
3. Use a `# type: ignore` or `# pyright: ignore[reportUnkonwnArgumentType]` comment to suppress the diagnostic.
```

**Comment by RobertCraigie** [comment]
Sure it's not unreasonable for it to generate a `reportUnknownArgumentType` but from my perspective it doesn't seem to be helpful.

In general I find the `reportUnknown*` diagnostics to be incredibly helpful in finding cases where the type checker doesn't have enough information to do its job but in this case it really doesn't matter what type the expression is inferred to be so what is the benefit in reporting it?
